src/target/target_probe.c

```diff
@@ -20,19 +20,34 @@
 #include "target_probe.h"
 
 #ifdef __APPLE__
-// https://bugs.llvm.org/show_bug.cgi?id=17775
-#define APPLE_STATIC
-// https://gcc.gnu.org/onlinedocs/cpp/Pragmas.html
-#define DO_PRAGMA_(x) _Pragma(#x)
-#define DO_PRAGMA(x)  DO_PRAGMA_(x)
-// __attribute__((alias)) is not supported in AppleClang.
-#define weak_alias(name, aliasname)     DO_PRAGMA(weak name = aliasname)
-#define CORTEXAR_PROBE_WEAK_NOP(name)   weak_alias(name, cortexar_probe_nop)
-#define CORTEXM_PROBE_WEAK_NOP(name)    weak_alias(name, cortexm_probe_nop)
-#define TARGET_PROBE_WEAK_NOP(name)     weak_alias(name, target_probe_nop)
-#define LPC55_DP_PREPARE_WEAK_NOP(name) weak_alias(name, lpc55_dp_prepare_nop)
+// __attribute__((alias)) is not supported in AppleClang, we need to define a
+// __attribute__((weak)) placeholder body that'll get pivoted by the linker.
+// if a "strong" definition is later available, the definition here will be
+// removed at linking time.
+// See:
+// - https://bugs.llvm.org/show_bug.cgi?id=17775
+// - "coalesced weak reference" in https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/MachOTopics/1-Articles/executing_files.html#//apple_ref/doc/uid/TP40001829-98432-TPXREF120
+#define CORTEXAR_PROBE_WEAK_NOP(name)                                        \
+	__attribute__((weak)) bool name(adiv5_access_port_s *u, target_addr_t v) \
```

Comment:
These should really maintain the proper parameter naming - `access_port` and `base_address` here, for example - to maintain code base style and not introduce more clang-tidy lints. Otherwise this looks great as a solution!